### PR TITLE
Add BNNR integration guide

### PR DESCRIPTION
**Pull request by mateuszwalo**
## Summary
Docs-only PR. Integration guide for BNNR (MIT): UltralyticsDetectionAdapter,
bbox-aware DetectionICD/DetectionAICD, BNNRTrainer reports. Does not replace yolo train.

## Runnable example
https://github.com/bnnr-team/bnnr/blob/v0.4.10/examples/integrations/ultralytics_yolo_quickstart.py

## Notes
- ICD/AICD figures load from bnnr-team/bnnr v0.4.10 on jsDelivr (Ultralytics .gitignore excludes *.png in docs).
- Tested quickstart with bnnr 0.4.10.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds a new Ultralytics integration guide for BNNR, documenting how to use bbox-aware XAI-guided augmentations, augmentation branch search, and structured reporting with YOLO through `UltralyticsDetectionAdapter`. 📚✨

### 📊 Key Changes
- Introduces a new docs page at `docs/en/integrations/bnnr.md` covering the external BNNR integration with Ultralytics YOLO.
- Documents BNNR concepts including `UltralyticsDetectionAdapter`, `BNNRTrainer`, `DetectionICD`, and `DetectionAICD` for bbox-aware augmentation workflows.
- Adds installation, quickstart, dataset format, limitations, FAQ, and licensing guidance for users evaluating the integration.
- Includes code snippets showing how to configure BNNR augmentations and initialize the adapter with Ultralytics weights.
- Adds BNNR to the integrations index in `docs/en/integrations/index.md`.
- Registers the new integration page in `mkdocs.yml` so it appears in the documentation navigation. 🧭

### 🎯 Purpose & Impact
- Expands Ultralytics documentation with a third-party integration option for users interested in XAI-guided detection augmentation experiments.
- Helps users understand that BNNR complements, rather than replaces, standard Ultralytics workflows like `yolo train` and `model.train()`.
- Makes the integration easier to discover through both the integrations landing page and docs navigation.
- Provides practical onboarding guidance that can reduce setup friction for researchers and advanced users exploring structured reports and augmentation search. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mateuszwalo, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you shortly. Please review the checklist below for a smooth integration of your documentation update 📚✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: mkdocs.yml [hunk 1]
```diff
@@ -594,6 +594,7 @@ nav:
       - Albumentations: integrations/albumentations.md
       - Amazon SageMaker: integrations/amazon-sagemaker.md
       - Axelera: integrations/axelera.md
+      - BNNR: integrations/bnnr.md
```
Comment (UltralyticsAssistant) [inline mkdocs.yml:624]: ⚠️ **HIGH**: This item is indented less than the surrounding integration entries, so it won't be nested under the same `Integrations` list and can break or misplace the MkDocs nav. Align it with the adjacent items.

**Suggested change:**
```suggestion
      - BNNR: integrations/bnnr.md
```

File: docs/en/integrations/bnnr.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    num_classes=80,
+)
+```
+
+## Dataset and image format
+
+Detection datasets for BNNR must provide:
+
+- **Images**: `torch.Tensor` `(C, H, W)` in **[0, 1]** float (do not pass 0-255 float tensors to YOLO through the adapter).
+- **Targets**: Ultralytics-style dicts with `boxes` and `labels`.
```
Comment (UltralyticsAssistant) [inline docs/en/integrations/bnnr.md]: 💡 **MEDIUM**: Describing `boxes`/`labels` as "Ultralytics-style" is misleading. Ultralytics detection batches use keys like `bboxes`, `cls`, and `batch_idx`, so readers may construct the wrong target format for the adapter if they follow this literally. Please either call this adapter-specific input or document the mapping explicitly.

**Suggested change:**
```suggestion
- **Targets**: Adapter-specific dicts with `boxes` and `labels` (these are not the same keys Ultralytics uses internally).
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Nice addition overall—the guide is structured well and the integration/index wiring is almost complete. The main things to fix are the misindented `mkdocs.yml` nav entry and the misleading dataset-format wording around "Ultralytics-style" target keys.

💬 Posted 2 inline comments

**Comment by mateuszwalo** [comment]
I have read the CLA Document and I sign the CLA

File: mkdocs.yml (hunk 1 above)
Comment (mateuszwalo) [reply mkdocs.yml:624]: BNNR uses the same indent as the other integration entries (6 spaces under Integrations). The suggested patch is a no-op on GitHub, so I resolved the thread.

**Comment by mateuszwalo** [comment]
Friendly bump on this one :D it's been open a few weeks and I want to make sure it isn't stuck

**Comment by glenn-jocher** [comment]
Thanks for the bump @mateuszwalo — it isn’t blocked; it’s awaiting maintainer review. Please keep the branch synced with `main` and CI green, and we’ll review the docs-only integration changes as bandwidth allows, especially around external dependency/version stability and long-term maintenance scope.

**Comment by glenn-jocher** [comment]
Hi @mateuszwalo, this branch is conflicting with `main` and we can't push to `bnnr-team/ultralytics` directly, so could you please apply the following?

1. `git merge main` — the only conflict is in `docs/en/integrations/index.md`, where `main` added an `AMD` entry in the same spot as `BNNR`. Keep both, in this order (matches `mkdocs.yml` nav):

```
- [AMD](amd.md): Train, validate, and run Ultralytics models on supported AMD GPUs with PyTorch ROCm, with a clear breakdown of MIGraphX, DirectML, and Ryzen AI NPU availability.

- [BNNR](bnnr.md): Add XAI-guided, bbox-aware augmentation search and training reports for YOLO via the external BNNR `UltralyticsDetectionAdapter` (Python API; does not replace `yolo train`).
```

2. In `docs/en/integrations/bnnr.md`, drop the last "Learn more" bullet — `https://github.com/bnnr-team/bnnr/blob/v0.4.10/docs/citation.md` 404s and would fail our link check.
3. In the same file, add `from bnnr import BNNRTrainer` to the augmentations snippet, which currently uses `BNNRTrainer` without importing it.
4. Run `npx prettier@3.8.5 --tab-width 4 --print-width 120 --write docs/en/integrations/bnnr.md` so the two tables match our docs formatting.

One review note: the page pins BNNR 0.4.10 throughout while PyPI is already on 0.7.1, so please confirm the pinned version and deep links are the ones you want to maintain long term.

Thanks!

**Comment by glenn-jocher** [comment]
Hi @mateuszwalo, thank you for putting together this integration guide — it's a well-written page and we genuinely appreciate BNNR's interest in building on Ultralytics YOLO. After review we've decided not to add a BNNR integration page to the official docs at this time, so we're closing this as not planned. That's a call about our docs scope, not about your work: please keep building the integration, and a guide hosted in BNNR's own docs (which we're happy to have linked from community resources) is a great home for it. If the integration matures and gains adoption, we'd welcome revisiting this. Keep up the great work! 🚀
